examples/ecs-target/main.tf

```diff
@@ -0,0 +1,325 @@
+        attach_role_arn = true
+        ecs_target = {
+          task_count          = 1
+          task_definition_arn = aws_ecs_task_definition.hello_world.arn
+        }
+      },
+      {
+        name = "send-putobject-events-to-cloudwatch"
+        arn  = aws_cloudwatch_log_group.this.arn
+      }
+    ]
+  }
+}
+
+locals {
+  name               = random_pet.this.id
+  ec2_resources_name = "${random_pet.this.id}-dev"
+  environment        = "dev"
+}
+
+##################
+# Extra resources
+##################
+
+resource "random_pet" "this" {
+  length = 2
+}
+
+resource "aws_cloudwatch_log_group" "this" {
+  name = "/aws/events/${random_pet.this.id}"
+
+  tags = {
+    Name = "${random_pet.this.id}-log-group"
+  }
+}
+
+#####
+# S3
+#####
+
+module "cloudtrail_bucket" {
+  source  = "terraform-aws-modules/s3-bucket/aws"
+  version = "~> 2.0"
+
+  bucket = "${random_pet.this.id}-cloudtrail"
+}
+
+resource "aws_s3_bucket_policy" "this" {
+  bucket     = module.cloudtrail_bucket.s3_bucket_id
+  depends_on = [module.cloudtrail_bucket]
+  policy     = data.aws_iam_policy_document.cloudtrail_policy.json
+}
+
+data "aws_iam_policy_document" "cloudtrail_policy" {
+  statement {
+    sid     = "AWSCloudTrailAclCheck20150319"
+    effect  = "Allow"
+    actions = ["s3:GetBucketAcl"]
+    resources = [
+      module.cloudtrail_bucket.s3_bucket_arn,
+      "${module.cloudtrail_bucket.s3_bucket_arn}/*"
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
+  }
+
+  statement {
+    sid     = "AWSCloudTrailWrite20150319"
+    effect  = "Allow"
+    actions = ["s3:PutObject"]
+    resources = [
+      "${module.cloudtrail_bucket.s3_bucket_arn}/AWSLogs/${data.aws_caller_identity.current.account_id}/*"
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
+
+    condition {
+      test     = "StringEquals"
+      variable = "s3:x-amz-acl"
+      values = [
+        "bucket-owner-full-control"
+      ]
+    }
+  }
+}
+
+module "source_bucket" {
+  source  = "terraform-aws-modules/s3-bucket/aws"
+  version = "~> 2.0"
+
+  bucket = "${random_pet.this.id}-source"
+}
+
+resource "aws_cloudtrail" "this" {
+  name                          = random_pet.this.id
+  s3_bucket_name                = module.cloudtrail_bucket.s3_bucket_id
+  include_global_service_events = true
+  is_multi_region_trail         = true
+
+  event_selector {
+    read_write_type           = "All"
+    include_management_events = true
+
+    data_resource {
+      type   = "AWS::S3::Object"
+      values = ["${module.source_bucket.s3_bucket_arn}/"]
+    }
+  }
+}
+
+data "aws_caller_identity" "current" {}
+
+######
+# ECS
+######
+
+module "ecs" {
+  source  = "terraform-aws-modules/ecs/aws"
+  version = "~> 3.0"
+
+  name = random_pet.this.id
+
+  capacity_providers = ["FARGATE", "FARGATE_SPOT", aws_ecs_capacity_provider.prov1.name]
```

Comment:
Please remove `aws_ecs_capacity_provider.prov1.name` from this example since it has a lot of nested dependencies (asg, instance profile, vpc, ...) and doesn't really have an impact on ECS+EventBridge integration we want to show in example.